tests/test_intl/test_intl.py

```diff
@@ -681,50 +682,188 @@ def test_translation_progress_classes_true(app):
+            # the colorama.ansitowin32 module, so we forcibly disable colors.
+            mock.setenv('NO_COLOR', '1')
+            # apply the patch only for Windows
+            clock = _MockWindowsClock()
+            mock.setattr('time.time_ns', clock.time)
+            # Use clock.sleep() to emulate time.sleep() but do not try
+            # to mock the latter since this might break other libraries.
+            mock.setattr('sphinx.util.i18n.CatalogInfo.write_mo', mock_write_mo)
+        yield mock, clock
+
+
 @sphinx_intl
-# use individual shared_result directory to avoid "incompatible doctree" error
-@pytest.mark.sphinx(testroot='builder-gettext-dont-rebuild-mo')
-def test_gettext_dont_rebuild_mo(make_app, app_params):
-    # --- don't rebuild by .mo mtime
-    def get_update_targets(app_):
-        app_.env.find_files(app_.config, app_.builder)
-        added, changed, removed = app_.env.get_outdated_files(config_changed=False)
-        return added, changed, removed
+# use the same testroot as 'gettext' since the latter contains less PO files
+@pytest.mark.sphinx('dummy', testroot='builder-gettext-dont-rebuild-mo', freshenv=True)
+def test_dummy_should_rebuild_mo(mock_time_and_i18n, make_app, app_params):
+    mock, clock = mock_time_and_i18n
+    assert os.name == 'posix' or clock.time() == 0
 
     args, kwargs = app_params
+    app = make_app(*args, **kwargs)
+    po_path, mo_path = _get_bom_intl_path(app)
+
+    # creation time of the those files (order does not matter)
+    bom_rst = app.srcdir / 'bom.rst'
+    bom_rst_time = time.time_ns()
+
+    index_rst = app.srcdir / 'index.rst'
+    index_rst_time = time.time_ns()
+    po_time = time.time_ns()
+
+    # patch the 'creation time' of the source files
+    assert _set_mtime_ns(po_path, po_time) == po_time
+    assert _set_mtime_ns(bom_rst, bom_rst_time) == bom_rst_time
+    assert _set_mtime_ns(index_rst, index_rst_time) == index_rst_time
+
+    assert not mo_path.exists()
+    # when writing mo files, the counter is updated by calling
+    # patch_write_mo which is called to create .mo files (and
+    # thus the timestamp of the files are not those given by
+    # the OS but our fake ones)
+    app.build()
+    assert mo_path.exists()
+    # Do a real sleep on POSIX, or simulate a sleep on Windows
+    # to ensure that calls to time.time_ns() remain consistent.
+    clock.sleep(0.1 if os.name == 'posix' else 1)
+
+    # check that the source files were not modified
+    assert bom_rst.stat().st_mtime_ns == bom_rst_time
+    assert index_rst.stat().st_mtime_ns == index_rst_time
+    # check that the 'bom' document is discovered after the .mo
+    # file has been written on the disk (i.e., read_doc() is called
+    # after the creation of the .mo files)
+    assert app.env.all_docs['bom'] > mo_path.stat().st_mtime
 
-    # phase1: build document with non-gettext builder and generate mo file in srcdir
```

Comment:
Let's either remove both the `phase1` and `phase2` comments, or neither of them (I'd prefer: keep them both).

Reply:
Ah, I think I forgot to write them again so I'll do it tomorrow